src/core/battle/tests.rs

```diff
@@ -1326,6 +1332,106 @@ fn throw_bomb_push() {
+            },
+        ],
+    );
+}
+#[test]
+fn throw_bomb_push_heavy() {
+    let prototypes = prototypes(&[
+        (
+            "thrower",
+            vec![
+                component_agent_one_attack(),
+                component_abilities(&[ability_throw_bomb_push()]),
+            ],
+        ),
+        (
+            "heavy",
+            [
+                component_agent_dull(),
+                component_strength(1),
+                component_blocker(Weight::Normal),
+            ]
+            .to_vec(),
+        ),
+        ("bomb_push", Vec::new()),
+    ]);
+    let scenario = scenario::default()
+        .object(P0, "thrower", PosHex { q: 0, r: 0 })
+        .object(P1, "heavy", PosHex { q: 1, r: 2 });
+    let mut state = debug_state(prototypes, scenario);
+    exec_and_check(
+        &mut state,
+        command::UseAbility {
+            id: Id(0),
+            pos: PosHex { q: 0, r: 2 },
+            ability: ability_throw_bomb_push(),
+        },
+        &[
+            Event {
+                active_event: event::UseAbility {
+                    id: Id(0),
+                    pos: PosHex { q: 0, r: 2 },
+                    ability: ability::BombPush {
+                        throw_distance: Distance(2),
+                    }
+                    .into(),
+                }
+                .into(),
+                actor_ids: vec![Id(0)],
+                instant_effects: vec![(
+                    Id(2),
+                    vec![
+                        effect::Create {
+                            pos: PosHex { q: 0, r: 0 },
+                            prototype: "bomb_push".into(),
+                            components: vec![
+                                component::Pos(PosHex { q: 0, r: 0 }).into(),
+                                component_meta("bomb_push"),
+                            ],
+                        }
+                        .into(),
+                        effect::Throw {
+                            from: PosHex { q: 0, r: 0 },
+                            to: PosHex { q: 0, r: 2 },
+                        }
+                        .into(),
+                    ],
+                )],
+                timed_effects: Vec::new(),
+                scheduled_abilities: vec![(
+                    Id(2),
+                    vec![PlannedAbility {
+                        rounds: 0,
+                        phase: Phase(0),
+                        ability: Ability::ExplodePush,
+                    }],
+                )],
+            },
+            Event {
+                active_event: event::UseAbility {
+                    id: Id(2),
+                    pos: PosHex { q: 0, r: 2 },
+                    ability: Ability::ExplodePush,
+                }
+                .into(),
+                actor_ids: vec![Id(2)],
+                instant_effects: vec![
+                    (
+                        Id(1),
+                        vec![effect::Knockback {
```

Comment:
It might be a good idea to post-check the resulting position of the object. Something like

```rust
assert_eq!(state.parts().pos.get(Id(1)).0, PosHex { q: 2, r: 2 });
```